apps/export/service.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import json
+from superdesk.services import BaseService
+from superdesk import get_resource_service
+from superdesk.errors import SuperdeskApiError
+from superdesk.publish.formatters import get_all_formatters
+from eve.validation import ValidationError
+from io import BytesIO
+from zipfile import ZipFile
+from flask import current_app as app
+
+logger = logging.getLogger(__name__)
+
+
+class ExportService(BaseService):
+    def create(self, docs, **kwargs):
+        doc = docs[0]
+        formatter = self._validate_and_get_formatter(doc)
+
+        validate = doc.get('validate', False)
+        archive_service = get_resource_service('archive')
+
+        unsuccessful_exports = 0
+        try:
+            in_memory_zip = BytesIO()
+            with ZipFile(in_memory_zip, 'a') as zip:
+                for item_id in doc.get('item_ids'):
+                    item = archive_service.find_one(req=None, _id=item_id)
+                    if item:
+                        try:
+                            contents = self.get_formatted_doc(item, formatter, validate)
+                            zip.writestr(item_id + '.txt', contents.encode("UTF-8"))
+                        except:
+                            unsuccessful_exports += 1
+                    else:
+                        unsuccessful_exports += 1
+
+            url = None
+            # Store the zip file on media_storage
+            # only if at least one item is formatted successfully
+            if unsuccessful_exports < len(doc.get('item_ids')):
+                zip_id = app.media.put(in_memory_zip.getvalue(), filename='export.zip', content_type='application/zip')
```

Comment:
I think that we need to add a celery task to clean those files after a given period of time.